src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java

```diff
@@ -0,0 +1,213 @@
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.Scope;
+import com.puppycrawl.tools.checkstyle.api.TextBlock;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.AnnotationUtil;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * <p>
+ * Checks for missing Javadoc comments for class, enum, interface, and annotation interface
+ * definitions. The scope to verify is specified using the {@code Scope} class and defaults
+ * to {@code Scope.PUBLIC}. To verify another scope, set property scope to one of the
+ * {@code Scope} constants.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code scope} - specify the visibility scope where Javadoc comments are checked.
+ * Default value is {@code public}.
+ * </li>
+ * <li>
+ * Property {@code excludeScope} - specify the visibility scope where Javadoc comments are not
+ * checked. Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code allowedAnnotations} - specify the list of annotations that allow missed
+ * documentation. Only short names are allowed, e.g.{@code Generated}. Default value is
+ * {@code Generated}.
+ * </li>
+ * <li>
+ * Property {@code tokens} - tokens to check Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">INTERFACE_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">CLASS_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">ENUM_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">ANNOTATION_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"/&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code private} scope:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="scope" value="private"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code private} classes only:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="scope" value="private"/&gt;
+ *   &lt;property name="excludeScope" value="package"/&gt;
```

Comment:
if target is private only .... what is reason or excludeScope=package ?
smth is missed in description of Check.

Reply:
@romani Maybe you are confused by what scope does? See http://checkstyle.sourceforge.net/property_types.html#scope
Private doesn't mean only private methods. 
> 'package' means all 'package', 'protected' and 'public'

So private would mean private and including everything package.

Reply:
We do have an issue to break away from scope and use access modifier.
See https://github.com/checkstyle/checkstyle/issues/3511

Reply:
Omh, .... Do you know definition of java scope ? I do not see any of at https://docs.oracle.com/javase/specs/jls/se8/html/

Reply:
Do you have energy to not use scope ? We should not leak such a weird stuff to public

Reply:
For this issue, I would prefer to just have these PRs split the checks into 2. It would be easier for compatibility as users could just copy existing config and change module name, for the most part. This is why I am doing a direct copy of the original check.

This check came from `JavadocType`, so it is already leaking to the public and it and other checks requires the same changes too. It is also not clear from other issue how we would change check to still use `anoninner` or if we would break away from it by dropping.

> Do you know definition of java scope ?

There are some mentions of scope in the JLS but not directed at what we are discussing here. ( https://docs.oracle.com/javase/specs/jls/se7/html/jls-6.html#jls-6.3 )
Previous commit says `Changed the classes Scope and SeverityLevel to be based on the Java 5 Enum class. Used patch #2004776 from Travis Schneeberger.` from 2008. Original commit says `javadoc checkscope implementation` from 2002.

Reply:
we need to explain users what this weird term mean, at least by examples.
Please add a bit more examples or some words to http://checkstyle.sourceforge.net/property_types.html#scope
If we have home made term, lets at least explain it.

Please do not to forget to add "MIGRATION NOTE" to issue after merge.

Reply:
Done.